decision-records/007-javascript-file-formats.md

```diff
@@ -0,0 +1,84 @@
+# JavaScript file names and formats
+
+## Decision
+
+We will consolidate all published JavaScript in the [`govuk-frontend` npm package](https://www.npmjs.com/package/govuk-frontend) into a single `govuk` directory.
+
+We will publish JavaScript in the following formats:
+
+1. ECMAScript (ES) modules
+2. ECMAScript (ES) modules, bundled
+3. Universal Module Definition (UMD), bundled
+
+We will use ECMAScript (ES) modules by default and only bundle JavaScript files that are considered "entry points" into GOV.UK Frontend, such as `all.mjs` and exported components (for example `accordion.mjs`).
+
+We will deprecate the Universal Module Definition (UMD) format in a future release.
+
+For consistency, we will clearly update our file extensions to:
+
+1. Add prefix `.bundle` to bundled JavaScript file extensions
+2. Add prefix `.min` to minified JavaScript file extensions
+
+We will add our [GitHub release](https://github.com/alphagov/govuk-frontend/releases) JavaScript to the [`govuk-frontend` npm package](https://www.npmjs.com/package/govuk-frontend).
```

Comment:
Think it's worth being clearer about what's being included from the package – as far as I recall it's just the minified JS?

Reply:
Sorry, our paths crossed and it was a post-sports day change to add the directory listing example 😊

Pushed up a bit more now